internal/gatewayapi/helpers.go

```diff
@@ -400,6 +406,10 @@ func irStringKey(gatewayNs, gatewayName string) string {
 	return fmt.Sprintf("%s/%s", gatewayNs, gatewayName)
 }
 
+func irMergedInfraPortName(listener *ListenerContext) string {
+	return fmt.Sprintf("%s-%s-%s", listener.Name, listener.gateway.Namespace, listener.gateway.Name)
+}
+
```

Comment:
will `irRouteName` and `irRouteDestinationName` collide since now there is nothing unique if the routes attach to multiple gateways resources which now will manifests into a single xds config
or can we reuse ?

Reply:
and you then want to do something similar to https://github.com/envoyproxy/gateway/blob/2ecc54da663dc10c9257a61843cd14ba1bb77dcc/internal/provider/utils/utils.go#L33C1-L33C1 in the infra layer side